src/Boot/DrupalBoot8.php

```diff
@@ -224,4 +225,22 @@ public function terminate()
             $this->kernel->terminate($this->getRequest(), $response);
         }
     }
+
+  /**
+   * Returns the factory method that can be used to retrieve the given kernel.
+   *
+   * @param string $kernel
+   *   The kernel to retrieve.
+   *
+   * @return callable
+   *   The factory method.
+   */
+    protected function getKernelFactory($kernel)
```

Comment:
This should probably be a method of Drush\Boot\Kernels.

Reply:
Ok! I put this in `DrupalBoot8` because these factory methods are Drupal 8-specific and might be different for Drupal 9 in the future. I agree that for the moment we only have one set of kernels, so we can put the code in `Kernels` instead.

Reply:
Done.

Reply:
You make a good point about Drupal 9. I like the separation provided by the `Kernels` class, though; I suppose if this needs to change for Drupal 9, then we'd just make a parallel subclass for Drupal 9 and instantiate it from DrupalBoot9.